python-lib/design/ab_dispatcher.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+import pandas as pd
+import logging
+
+from design.constants import AttributionMethod, Group
+
+logger = logging.getLogger()
+logging.basicConfig(level=logging.INFO, format="AB testing plugin %(levelname)s - %(message)s")
+
+
+class AbDispatcher(object):
+
+    def __init__(self, size_A: int, size_B: int, reference_column: str):
+        self.size_A = size_A
+        self.size_B = size_B
+        self.reference_column = reference_column
+
+    def dispatch(self, experiment_population: np.array, attribution: AttributionMethod, input_df: pd.DataFrame) -> tuple:
+        """Dispatch the experiment population into two groups
+
+        :param np.array experiment_population: array containing references to the experiment population
+        :param AttributionMethod attribution: defines how to deal with the leftover users.
```

Comment:
done in https://github.com/dataiku/dss-plugin-ab-testing/pull/1/commits/42e6adc2eb62d94294976c37afbd072cbbf53120

Reply:
for the name attribution, I don't know what to pick, leftover_management? manage_leftover?

Reply:
replaced attribution with leftovers_management in https://github.com/dataiku/dss-plugin-ab-testing/pull/1/commits/71c913edcddf909a14b0b1f80d9fadee2478b515

Reply:
replaced leftovers_management with leftovers_handling in https://github.com/dataiku/dss-plugin-ab-testing/pull/1/commits/82c0423fdbf9d686ea29802492947db871085d93 , also changed the labels in the recipe's settings (instead of attribution for the leftover users , it's now deal with leftover users)